src/test/java/io/supertokens/test/passwordless/api/PasswordlessConsumeCodeAPITest2_10.java

```diff
@@ -0,0 +1,341 @@
+            return;
+        }
+
+        String email = "test@example.com";
+        CreateCodeResponse createResp = Passwordless.createCode(process.getProcess(), email, null, null, null);
+        Thread.sleep(1500);
+
+        JsonObject consumeCodeRequestBody = new JsonObject();
+        consumeCodeRequestBody.addProperty("deviceId", createResp.deviceId);
+        consumeCodeRequestBody.addProperty("userInputCode", createResp.userInputCode);
+
+        JsonObject response = HttpRequestForTesting.sendJsonPOSTRequest(process.getProcess(), "",
+                "http://localhost:3567/recipe/signinup/code/consume", consumeCodeRequestBody, 1000, 1000, null,
+                Utils.getCdiVersion2_10ForTests(), "passwordless");
+
+        assertEquals("EXPIRED_USER_INPUT_CODE", response.get("status").getAsString());
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void testIncorrectUserInputCode() throws Exception {
+        String[] args = { "../" };
+
+        Utils.setValueInConfig("passwordless_max_code_input_attempts", "2"); // Only 2 code entries permitted (1 retry)
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        String email = "test@example.com";
+        CreateCodeResponse createResp = Passwordless.createCode(process.getProcess(), email, null, null, null);
+
+        JsonObject consumeCodeRequestBody = new JsonObject();
+        consumeCodeRequestBody.addProperty("deviceId", createResp.deviceId);
+        consumeCodeRequestBody.addProperty("userInputCode", createResp.userInputCode + "nope");
+
+        {
+            JsonObject response = HttpRequestForTesting.sendJsonPOSTRequest(process.getProcess(), "",
+                    "http://localhost:3567/recipe/signinup/code/consume", consumeCodeRequestBody, 1000, 1000, null,
+                    Utils.getCdiVersion2_10ForTests(), "passwordless");
+
+            assertEquals("INCORRECT_USER_INPUT_CODE", response.get("status").getAsString());
+        }
+
+        {
+            JsonObject response = HttpRequestForTesting.sendJsonPOSTRequest(process.getProcess(), "",
+                    "http://localhost:3567/recipe/signinup/code/consume", consumeCodeRequestBody, 1000, 1000, null,
+                    Utils.getCdiVersion2_10ForTests(), "passwordless");
+
+            assertEquals("RESTART_FLOW_ERROR", response.get("status").getAsString());
+        }
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    private void checkResponse(JsonObject response, Boolean isNewUser, String email, String phoneNumber) {
```

Comment:
You need to also check that preAuthSessionId is in the response, and check that the number of items in the response are as expected. In general, make sure to follow the CDI spec always when writing these tests.